query-node/mappings/src/workingGroups.ts

```diff
@@ -709,74 +721,110 @@ export async function workingGroups_StatusTextChanged({ store, event }: EventCon
   await store.save<StatusTextChangedEvent>(statusTextChangedEvent)
 }
 
-export async function workingGroups_LeadRemarked({ store, event }: EventContext & StoreContext): Promise<void> {
-  const [metadataByte] = new WorkingGroup_LeadRemarkedEvent_V1001(event).params
-  const group = await getWorkingGroupOrFail(store, event)
-
-  const metadata = deserializeMetadata(RemarkMetadataAction, metadataByte)
-  if (metadata?.moderatePost) {
+async function processWorkingGroupsRemark(
+  store: DatabaseManager,
+  event: SubstrateEvent,
+  group: WorkingGroup,
+  decodedMetadata: DecodedMetadataObject<IRemarkMetadataAction> | null,
+  isLead = false,
+  workerId: number
+): Promise<any> {
+  // Moderate Post
+  if (decodedMetadata?.moderatePost) {
     if (group.name !== 'forumWorkingGroup') {
       return invalidMetadata(`The ${group.name} is incompatible with the remarked moderatePost`)
     }
-    const { postId, rationale } = metadata.moderatePost
-    const actor = await getWorkingGroupLeadOrFail(store, group.name)
+
+    const { postId, rationale } = decodedMetadata.moderatePost
+    const actor = isLead
+      ? await getWorkingGroupLeadOrFail(store, group.name)
+      : await getWorkerOrFail(store, group.name, workerId)
 
     const post = await getById(store, ForumPost, postId)
     if (!post) {
       return invalidMetadata(`Forum post not found by id: ${postId}`)
     }
-    await moderatePost(store, event, 'leadRemark', post, actor, rationale)
-  } else if (metadata?.verifyValidator) {
+    return await moderatePost(store, event, 'leadRemark', post, actor, rationale)
+  }
+
+  if (decodedMetadata?.verifyValidator) {
     if (group.name !== 'membershipWorkingGroup') {
       return invalidMetadata(`The ${group.name} can't verify the validator's membership`)
     }
-    const { memberId, isVerified } = metadata.verifyValidator
+    const { memberId, isVerified } = decodedMetadata.verifyValidator
 
     const member = await getById(store, Membership, memberId)
     if (!member) {
       return invalidMetadata(`Membership not found by id: ${memberId}`)
     }
     member.metadata.isVerifiedValidator = isVerified
     await store.save<MemberMetadata>(member.metadata)
-    await store.save<Membership>(member)
-  } else {
-    return invalidMetadata('Unrecognized remarked action')
+    return await store.save<Membership>(member)
+  }
+
+  if (decodedMetadata?.createTag) {
```

Comment:
I suggest to limit tag-management to Forum Working Group only. I don't see a need for other WGs to manage this tbh. Also the permissions you have currently only check worker ID which is duplicated across WGs.

Reply:
Yes. I will update the code